File: trunk/classes/App/Model/OrderAddress.php
<?php

namespace App\Model;

/**
 * Class OrderAddress
 * @package App\Model
 * @property int $id
 * @property int $country_id
 * @property int $customer_id
 * @property int $order_id
 * @property string $full_name
 * @property string $address_line_1
 * @property string $address_line_2
 * @property string $city
 * @property string $region
 * @property string $zip
 * @property string $phone
 * @property string $address_type
 */
class OrderAddress extends BaseModel {

    public $table = 'tbl_order_address';
    public $id_field = 'id';

    public function getShippingAddress()
    {
        $addresses = $this->find_all()->as_array();
        foreach ($addresses as $address) {
            if ($address->address_type == 'shipping') {
                return $address;
            }
        }
        return array();
    }

    public function getBillingAddress()
    {
        $addresses = $this->find_all()->as_array();
        foreach ($addresses as $address) {
            if ($address->address_type == 'billing') {
                return $address;
            }
        }
        return array();
    }
}

File: trunk/subprojects/android/hackazon/src/com/ntobjectives/hackazon/model/OrderAddress.java
package com.ntobjectives.hackazon.model;

import java.util.ArrayList;

/**
 * Created with IntelliJ IDEA by Nick Chervyakov.
 * User: Nikolay Chervyakov
 * Date: 28.10.2014
 * Time: 11:39
 */
@SuppressWarnings("serial")
public class OrderAddress {
    public int id;
    public String full_name;
    public String address_line_1;
    public String address_line_2;
    public String city;
    public String region;
    public String zip;
    public String country_id;
    public String phone;
    public Integer customer_id;
    public String address_type;
    public Integer order_id;

    public OrderAddress() {
    }

    @SuppressWarnings("serial")
    public static class List extends ArrayList<OrderAddress> {
    }

    public static class OrderAddressesResponse extends CollectionResponse<List> {
    }

    public void fillFromCustomerAddress(CustomerAddress address) {
        address_line_1 = address.address_line_1;
        address_line_2 = address.address_line_2;
        city = address.city;
        country_id = address.country_id;
        zip = address.zip;
        phone = address.phone;
        region = address.region;
        full_name = address.full_name;
        customer_id = address.customer_id;
    }
}
